app/src/main/java/com/zeapo/pwdstore/UserPreference.kt

```diff
@@ -163,16 +157,21 @@ class UserPreference : AppCompatActivity() {
 
             appVersionPreference?.summary = "Version: ${BuildConfig.VERSION_NAME}"
 
-            keyPreference?.onPreferenceClickListener = ClickListener {
-                val providerPackageName = requireNotNull(sharedPreferences.getString("openpgp_provider_list", ""))
-                if (providerPackageName.isEmpty()) {
-                    Snackbar.make(requireView(), resources.getString(R.string.provider_toast_text), Snackbar.LENGTH_LONG).show()
-                    false
-                } else {
-                    val intent = Intent(callingActivity, PgpActivity::class.java)
-                    intent.putExtra("OPERATION", "GET_KEY_ID")
-                    startActivityForResult(intent, IMPORT_PGP_KEY)
-                    true
+            keyPreference?.let { pref ->
+                updateKeyIDsSummary(pref)
+                pref.onPreferenceClickListener = ClickListener {
+                    val providerPackageName = requireNotNull(sharedPreferences.getString("openpgp_provider_list", ""))
+                    if (providerPackageName.isEmpty()) {
+                        Snackbar.make(requireView(), resources.getString(R.string.provider_toast_text), Snackbar.LENGTH_LONG).show()
+                        false
+                    } else {
+                        val intent = Intent(callingActivity, GetKeyIdsActivity::class.java)
+                        val keySelectResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+                            updateKeyIDsSummary(pref)
```

Comment:
Thanks, the selected key ID not showing up in the summary right after selection has caused me some headaches in the past.